### PR TITLE
[Spot, Margin] - Replace RPC usage for events spot & margin pages

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.monday.com/boards/2218344956/pulses/2731931003?doc_id=2731946223
https://sovryn.atlassian.net/browse/SOV-225

### CODE REVIEW

File: src/app/pages/MarginTradePage/hooks/useGetLimitOrderEvents.ts [hunk 1]
```diff
@@ -29,12 +28,9 @@ export const useGetLimitOrderEvents = () => {
...
   );
 
-  const { events } = useGetContractPastEvents(
-    'settlement',
-    'MarginOrderFilled',
-  );
+  const { data } = useGetMarginLimitOrderFilled();
 
-  useLog('LimitOrderTables', events);
+  const events = useMemo(() => data?.marginOrderFilleds, [data]);
```
Comment (tiltom) [inline src/app/pages/MarginTradePage/hooks/useGetLimitOrderEvents.ts:33]: Pls ask Betsy/Remy to correct the typo, should be `marginOrderFilled`.

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/app/pages/SpotTradingPage/components/LimitOrderTables/index.tsx [hunk 2]
```diff
@@ -26,31 +28,21 @@ export const LimitOrderTables: React.FC<ILimitOrderTablesProps> = ({
...
-  );
-
   return (
     <>
       <div className={classNames({ 'tw-hidden': activeTab !== 1 })}>
         <OpenPositionsTable
           orders={limitOrders.filter(item => item.filledAmount === '0')}
           loading={loading}
-          orderCreatedEvents={orderCreatedEvents.events}
+          orderCreatedEvents={orderCreatedEvents?.orderCreateds}
```
[the remark this answers is not in the document]
Comment (pietro-maximoff) [reply src/app/pages/SpotTradingPage/components/LimitOrderTables/index.tsx:37]: the same here @BetsyBraddock

File: src/app/pages/MarginTradePage/hooks/useGetLimitOrderEvents.ts (hunk 1 above)
Comment (BetsyBraddock) [reply src/app/pages/MarginTradePage/hooks/useGetLimitOrderEvents.ts:33]: Hi guys, the singular/plural naming is a graphql thing. If you want multiple MarginOrderFilled entities, they are called marginOrderFilled and you have to provide an ID. If you want multiple entities, they are called marginOrderFilleds

File: src/app/pages/SpotTradingPage/components/LimitOrderTables/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (BetsyBraddock) [reply src/app/pages/SpotTradingPage/components/LimitOrderTables/index.tsx:37]: See above comment about plurals in graphql

File: src/app/pages/MarginTradePage/hooks/useGetLimitOrderEvents.ts (hunk 1 above)
Comment (pietro-maximoff) [reply src/app/pages/MarginTradePage/hooks/useGetLimitOrderEvents.ts:33]: Thanks Betsy! @tiltom is it ok for you?

Comment (tiltom) [reply src/app/pages/MarginTradePage/hooks/useGetLimitOrderEvents.ts:33]: well yeah, it looks a bit weird but I understand the explanation :D

Comment (soulBit) [reply src/app/pages/MarginTradePage/hooks/useGetLimitOrderEvents.ts:33]: I hate this too but appreciate the fact you can't change it 😂

**Review by soulBit — DISMISSED** [review]

**Review by tiltom — DISMISSED** [review]

**Review by Alitasovryn — DISMISSED** [review]

File: src/app/pages/MarginTradePage/hooks/useGetLimitOrderEvents.ts (hunk 1 above)
Comment (creed-victor) [inline src/app/pages/MarginTradePage/hooks/useGetLimitOrderEvents.ts:33]:
```suggestion
  const events = useMemo(() => data?.marginOrderFilleds || [], [data]);
```

**Review by creed-victor — APPROVED** [review]

**Review by soulBit — APPROVED** [review]

**Comment by Alitasovryn** [comment]
Working well

**Pull request merged**